acra/guides/integrating-acra-translator-into-new-infrastructure/_index.md

```diff
@@ -4,55 +4,123 @@ bookCollapseSection: true
+it should be base64-encoded and passed to Acra services in `ACRA_MASTER_KEY` environment variable.
+
+Like this: `ACRA_MASTER_KEY="$(cat /tmp/master_key | base64)" acra-server ...`
+
+The second key is responsible for data encryption.
+There are actually more kinds of keys, read more about that on
+[Acra keys inventory]({{< ref "acra/security-controls/key-management/inventory.md" >}}).
+
+It is also possible to store keys in a Redis database, see
+[Scalable KV storages]({{< ref "acra/configuring-maintaining/key-storing/kv-stores.md#redis" >}}).
+
+### Note about Client ID
+
+When generating a key, you will always have to bind it with a [Cliend ID]({{< ref "client_id.md" >}}) or Zone ID.
+AcraTranslator distinguishes requests by [Client ID]({{< ref "client_id.md" >}}) and uses different encryption keys for 
+different client ID.
 
 ## Architecture and DataFlow of AcraTranslator-based infrastructure
 
 ![](/files/data-flow/acra-archi-translator-writer.png)
 
 Acra design values simplicity as much as security.
 
-1. The application sends [AcraStructs](/acra/acra-in-depth/data-structures/acrastruct) or [AcraBlocks](/acra/acra-in-depth/data-structures/acrablock) via HTTP or gRPC API.
-2. AcraConnector sends that request to AcraTranslator using [Themis Secure Session](/themis/crypto-theory/cryptosystems/secure-session/) (socket protection protocol).
-3. AcraTranslator accepts the request and attempts to decrypt an [AcraStruct](/acra/acra-in-depth/data-structures/acrastruct) or [AcraBlock](/acra/acra-in-depth/data-structures/acrablock). If the decryption is successful, it sends the resulting plaintext in the response. If decryption fails, AcraTranslator sends out a decryption error.
-4. AcraTranslator returns the data to AcraConnector (via [Themis Secure Session](/themis/crypto-theory/cryptosystems/secure-session/)), which in turn returns it to the application.
+1. The application sends [AcraStructs] or [AcraBlocks] via HTTP or gRPC API.
+2. AcraConnector sends that request to AcraTranslator using [Themis Secure Session] (socket protection protocol).
+3. AcraTranslator accepts the request and attempts to decrypt an [AcraStruct] or [AcraBlock]. If the decryption is
+   successful, it sends the resulting plaintext in the response. If decryption fails, AcraTranslator sends out a
+   decryption error.
+4. AcraTranslator returns the data to AcraConnector (via [Themis Secure Session]), which in turn returns it to the application.
 
-AcraTranslator reads decryption keys from key folder and stores them encrypted in memory. It uses LRU cache to increase performance by keeping only actively used keys in memory. The size of LRU cache can be configured depending on your server's load.
+AcraTranslator reads decryption keys from key folder and stores them encrypted in memory. It uses LRU cache to increase
+performance by keeping only actively used keys in memory. The size of LRU cache can be configured depending on your
```

Comment:
extra new line

Reply:
it's rendered as one piece of text. I split it into 120 character rows for easier reading and working with 2 columns of text (2 tabs in editor). For example python has the convention to split long rows into 80 ch rows. But in 2021 with 1920px screens we can use 120 (default in intellij idea), which is 1920 / 2 (columns) / 8 (pixels per character).
You can check it by opening this file as whole file and look how github renders it - https://github.com/cossacklabs/product-docs/blob/6e8d8bf62a4c89a2b8a87e90822eb115a4b5fe0c/acra/guides/integrating-acra-translator-into-new-infrastructure/_index.md

look how it looks:
![image](https://user-images.githubusercontent.com/6237700/146010482-2a6105c0-ec47-4872-abb8-5286925ca9c9.png)